pf/internal/convert/encoding.go

```diff
@@ -51,53 +48,45 @@ func (e *encoding) NewConfigEncoder(configType tftypes.Object) (Encoder, error)
 
 func (e *encoding) NewResourceEncoder(resource string, objectType tftypes.Object) (Encoder, error) {
 	mctx := newResourceSchemaMapContext(resource, e.SchemaOnlyProvider, e.ProviderInfo)
-	propertyEncoders, err := buildPropertyEncoders(mctx, objectType)
+	enc, err := NewObjectEncoder(mctx.schemaMap, mctx.schemaInfos, objectType)
 	if err != nil {
-		return nil, fmt.Errorf("cannot derive an encoder for resource %q: %w", resource, err)
-	}
-	enc, err := newObjectEncoder(objectType, propertyEncoders, mctx)
-	if err != nil {
-		return nil, fmt.Errorf("cannot derive an encoder for resource %q: %w", resource, err)
+		return nil, fmt.Errorf("cannot derive an encoder for resource %q: %w",
+			resource, err)
 	}
 	return enc, nil
 }
 
 func (e *encoding) NewResourceDecoder(resource string, objectType tftypes.Object) (Decoder, error) {
 	mctx := newResourceSchemaMapContext(resource, e.SchemaOnlyProvider, e.ProviderInfo)
-	propertyDecoders, err := buildPropertyDecoders(mctx, objectType)
-	if err != nil {
-		return nil, fmt.Errorf("cannot derive an decoder for resource %q: %w", resource, err)
-	}
-	propertyDecoders["id"] = newStringDecoder()
-	dec, err := newObjectDecoder(objectType, propertyDecoders, mctx)
+	dec, err := NewObjectDecoder(mctx.schemaMap, mctx.schemaInfos, objectType)
+	// Tests pass without this line. Was this intentional? Perhaps to support resources with
```

Comment:
This one change requires some scrutiny. Its seemed that currently the tests pass without it but there must have been reason to put this in - in the first place. I've not been enlightened by reading the commit log. Perhaps there is something about resources that have non-string IDs or are lacking an "id" field entirely, which might be possible in PF?

Reply:
Thinking possibly related to https://github.com/pulumi/pulumi-terraform-bridge/pull/1623 - this is something I would like to test here or ensure it is tested so it still works.